Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2949,9 +2954,43 @@ private void initializeSession(BranchReferralInitListener callback) {
             Log.i("BranchSDK", "Branch Warning: You are using your test app's Branch Key. Remember to change it to live Branch Key during deployment.");
         }
 
+        if (!prefHelper_.getExternalIntentUri().equals(PrefHelper.NO_STRING_VALUE)) {
+            registerAppInit(callback);
+        } else {
+            // Check if opened by facebook with deferred install data
+            boolean appLinkRqSucceeded = false;
+            deferredInitListener_ = new WeakReference<>(callback);
+            appLinkRqSucceeded = DeferredAppLinkDataHandler.fetchDeferredAppLinkData(context_, new DeferredAppLinkDataHandler.AppLinkFetchEvents() {
+                @Override
+                public void onAppLinkFetchFinished(String nativeAppLinkUrl) {
+                    if (nativeAppLinkUrl != null) {
+                        Uri appLinkUri = Uri.parse(nativeAppLinkUrl);
+                        String bncLinkClickId = appLinkUri.getQueryParameter(Defines.Jsonkey.LinkClickID.getKey());
+                        if (!TextUtils.isEmpty(bncLinkClickId)) {
+                            prefHelper_.setLinkClickIdentifier(bncLinkClickId);
+                            prefHelper_.setIsAppLinkTriggeredInit(true);
+                        }
+                    }
+                    BranchReferralInitListener deferredCallback = null;
+                    if (deferredInitListener_ != null) {
+                        deferredCallback = deferredInitListener_.get();
+                    }
+                    registerAppInit(deferredCallback);
```

Comment:
I have already tested with no internet condition and it seems it called back with in 500 ms.